### ISSUE TITLE
Remove ip_registration_newsletter from class Customer

### ISSUE THREAD

**Issue by Traumflug** [thirtybees/thirtybees#1064]
Noticed by #1056.

As customers often have dynamic IP addresses and also often use multiple devices, it doesn't make much sense to record the IP address used while setting up an account. As far as I can see, this IP address isn't used anywhere anyways, it just gets stored in a number of places.

Delivering an empty string during the deprecation period should be fine, like example customers stored during installation do.

Not recording the IP address also removes a topic for GDPR compliance.

**Comment by therampagerado** [comment]
I would have to say that getting the user IP when an order is made is mandatory in some cases. What if the card is stolen and they try to order product just to steal the money. All fraud measures are passed but the bank/police wants information for the customer.

At the moment I'm using a module to track those IPs and show them in the order page together with location for the ip. It would be nice to see this info in BO order page by default.

**Comment by Traumflug** [comment]
Well, this field doesn't store the IP used when ordering, but the one used during registration.

**Comment by therampagerado** [comment]
If this idea is to remove the stored IP during reg it's OK because this is not important. But the IP should be recorded when an order is made for sure. And if it can be shown in BO easily it would be nice too.

**Comment by eschiendorfer** [comment]
Yeah I agree, that saving the ip_address for newsletter registration doesn't make much sense these days.

**Comment by seigieu** [comment]
Just came across it by accident. Actually it makes sense in case someone entered someone else's email. And you would be accused of SPAM. Then you know what IP request came from.
